### PR TITLE
Sign In

### PR DESCRIPTION
**Pull request by CristianRWolox**
## Summary

- Make the endpoint `/users/sessions`
- Create new middleware (`credentials-match`) to validate the params.

## Trello Card

[Sign In](https://trello.com/c/xsugSLje/2-sign-in)

### CODE REVIEW

File: app/controllers/auth/sign-in.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const jwt = require('jsonwebtoken');
+const logger = require('../../logger');
+
+exports.signIn = (req, res, next) => {
+  try {
+    const { email, password } = req.body;
+    const token = jwt.sign({ email, password }, process.env.TOKEN_KEY, {
+      expiresIn: '1h'
+    });
```
Comment (dbenavidesv) [inline app/controllers/auth/sign-in.js:9]: put this on a function on a helper or utils folder

File: app/middlewares/credentials-match.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const bcrypt = require('bcrypt');
+const { conflictError, databaseError, EMAIL_DOES_NOT_EXIST, WRONG_PASSWORD } = require('../errors');
+const { User } = require('../models');
+
+const credentialsMatch = async (req, _res, next) => {
+  try {
+    const user = await User.findOne({ where: { email: req.body.email } });
+    if (user) {
+      const match = await bcrypt.compare(req.body.password, user.password);
+      return match ? next() : next(conflictError('password schema', WRONG_PASSWORD));
```
Comment (dbenavidesv) [inline app/middlewares/credentials-match.js:10]: This is not a conflict error, you should create a new error end use `401 Unauthorized`

File: app/middlewares/credentials-match.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    }
+    return next(conflictError('email schema', EMAIL_DOES_NOT_EXIST));
```
Comment (dbenavidesv) [inline app/middlewares/credentials-match.js:12]: Same here

File: app/middlewares/credentials-match.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  } catch {
+    return next(databaseError('database schema'));
```
Comment (dbenavidesv) [inline app/middlewares/credentials-match.js:14]: I don't understand this error here

**Review by dbenavidesv — APPROVED** [review]

File: app/middlewares/credentials-match.js (hunk 4 above)
Comment (CristianRWolox) [reply app/middlewares/credentials-match.js:14]: Change it to validationError, maybe that is more appropriate

**Pull request merged**